overlord/snapstate/handlers.go

```diff
@@ -910,7 +913,10 @@ func (m *SnapManager) undoUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	snapst.Active = true
-	reboot, err := m.backend.LinkSnap(oldInfo, deviceCtx, svcsToDisable, perfTimings)
+	linkCtx := backend.LinkContext{
+		PrevDisabledServices: svcsToDisable,
```

Comment:
should this have a FirstInstall: false for clarity?

Reply:
Added